### ISSUE TITLE
problem installing, dependency resolution

### ISSUE THREAD

**Issue by RobinKnipe** [oven-sh/bun#8751]
### What version of Bun is running?

1.0.26

### What platform is your computer?

Linux 5.15.0-92-generic x86_64 x86_64

### What steps can reproduce the bug?

- install bun using `sudo npm install -g bun`
- create a new svelte app `bun create svelte@latest`
- run `bun i` for first time dependency installation

### What is the expected behavior?

Something similar to `npm i`, which warns about a problem with a dependency, but still completes the installation process:

```shell
$ npm i
npm WARN ERESOLVE overriding peer dependency

added 228 packages, and audited 229 packages in 41s
```

### What do you see instead?

The install fails with an error:

```shell
$ bun i
bun install v1.0.26 (c75e768a)
warn: incorrect peer dependency "svelte@5.0.0-next.50"

error: No version matching ">=0.34.0-next.4 <1.0.0" found for specifier "svelte-eslint-parser" (but package exists)
```

### Additional information

I've tried manually making changes to the version values in the `package.json` and rerunning `bun i`, but even specifying the versions listed in the warning/error output just produces the same result.

**Comment by Super-Kenil** [comment]
I had installed bun in my linux machin using `curl -fsSL https://bun.sh/install | bash`, I too am facing this issue. For installing packages, I used `bun i` in the terminal. And its stuck at the following place since forever

`user>  bun i`
`0.04ms] ".env"`
`bun install v1.0.26 `
`  🔍 Resolving... `

**Comment by Super-Kenil** [comment]
@RobinKnipe , The issue seems to fixed. I checked it on multiple projects and devices. 

You can close the issue after checking it yourself

**Comment by RobinKnipe** [comment]
@Super-Kenil  I think the problem you were experiencing was different :shrug:  Certainly I'm still seeing the same problem and I appear to be using the latest version of bun:

```shell
$ bun i
bun install v1.0.26 (c75e768a)
warn: incorrect peer dependency "svelte@5.0.0-next.50"

error: No version matching ">=0.34.0-next.4 <1.0.0" found for specifier "svelte-eslint-parser" (but package exists)

$ sudo bun i -g bun
bun add v1.0.26 (c75e768a)

 installed bun@1.0.26 with binaries:
  - bun
  - bunx

 3 packages installed [4.45s]

$ bun --version
1.0.26

$ bun i
bun install v1.0.26 (c75e768a)
warn: incorrect peer dependency "svelte@5.0.0-next.50"

error: No version matching ">=0.34.0-next.4 <1.0.0" found for specifier "svelte-eslint-parser" (but package exists)
```

Specifically I believe the problem here is bun's failure (or choice?) to complete the process (even with problems). By contrast npm lists the problems but still creates the installation resources (`node_modules` & `package-lock.json`), so at least development can continue...

I don't even seem to be able to request the dependency to be resolved through installation:

```shell
$ bun i -D svelte-eslint-parser
bun add v1.0.26 (c75e768a)
warn: incorrect peer dependency "svelte@5.0.0-next.50"

error: No version matching ">=0.34.0-next.4 <1.0.0" found for specifier "svelte-eslint-parser" (but package exists)
```

It fails with the same message, but so fast that I suspect the response maybe cached in some way (hopefully that's just my previous experience with `lerna` tooling clouding my expectations of _the Brave New World_ :crossed_fingers:)

**Comment by GrygrFlzr** [comment]
I think this is because bun does not support *version ranges*.
`eslint-plugin-svelte` depends on `svelte-eslint-parser`, but declares it as a range in`package.json`:
```jsonc
  "dependencies": {
    // ...
    "svelte-eslint-parser": ">=0.34.0-next.4 <1.0.0"
  },
```
So bun cannot figure out what version to install.

As a workaround, you can use the `overrides` field in `package.json`:
```json
  "overrides": {
    "svelte-eslint-parser": "next"
  },
```
But imo this is a bug because bun should be able to handle this as npm and pnpm does.

**Comment by dylan-conway** [comment]
this semver bug is fixed in #9026
